### DISCUSSION TITLE
Configuration SSO with Azure AD

### DISCUSSION THREAD

**Discussion by IT-META** [inventree/InvenTree#8448] [Q&A] [answered]
Hi community,

I want to configure my fresh inventree installation with Azure AD SSO. I read the documentation and add the line in my .env file :

 social_backends:
  allauth.socialaccount.providers.microsoft

But when i connect to admin console, I can't add any Provider !

Thanks for your help

**Comment by SchrodingersGat** [comment]
> But when i connect to admin console, I can't add any Provider !

The admin console in InvenTree, you mean? Can you please provide some screenshots showing what you are trying to do

**Comment by IT-META** [comment]
Hello,

Yes it is since the inventree admin console 

![image](https://github.com/user-attachments/assets/3f103fe3-188c-4554-8c27-891eec7cdf32)

**Comment by SchrodingersGat** [comment, marked as answer]
> I read the documentation and add the line in my .env file 

Firstly, this should be in your `config.yaml` file (not the `.env` file). And secondly the format is wrong.

From [the docs page](https://docs.inventree.org/en/stable/settings/SSO/?h=#enable-provider-backends) the format (in your `config.yaml` file) should look like this:

```yaml
social_backends:
  - allauth.socialaccount.providers.microsoft
```

If you look at the [default configuration file](https://docs.inventree.org/en/stable/start/config/#configuration-file-template) it has it already setup (except it is commented out)

**Comment by IT-META** [comment]
Hello,

I have this error now : **validating /etc/inventree/docker-compose.yml: (root) Additional property social_backends is not allowed**

**Reply by SchrodingersGat** [reply]
Can you please provide the full error? It is probably a yaml formatting issue - use the config template file (linked above) as an example

**Comment by IT-META** [comment]
Hello,
It works !
I didn't configure the right file....
Thanks for your help.

**Comment by IT-META** [comment]
I have another problem yet. It seems that the CallbackUri that I have configured on the tenant is not the good. I have this error:
AADSTS50011: The redirect URI 'https://inventree.intra.test.com/accounts/microsoft/login/callback/' specified in the request does not match the redirect URIs configured for the application 'b127666d-ed56-5532-9834-54re38v154'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. Can I have to create a new ticket ?

**Reply by matmair** [reply]
That is a config error on Entra side, probably in the callback set there.
We can not really help you with your Entra ID setup, that is very dependent on your tenant policies.
